File: 2.1/Counter/Counter/Clock.cs
using System;

namespace Counter
{
    public class Clock
    {
        private Counter _seconds;
        private Counter _minutes;
        private Counter _hours;
        public Clock()
        {

            _seconds = new Counter("seconds");
            _minutes = new Counter("minutes");
            _hours = new Counter("hours");

        }

        public void Reset()
        {
            _seconds.Reset();
            _minutes.Reset();
            _hours.Reset();

        }
        public string ReadTime
        {
            get
            {
                return $"{_hours.Count}:{_minutes.Count}:{_seconds.Count}";
            }
        }
        public void Tick()
        {
            if (_seconds.Count < 59)
            {
                _seconds.Increment();

            } else
            {
                _seconds.Reset();
                if (_minutes.Count < 59)
                {
                    _minutes.Increment();
                } else
                {
                    _minutes.Reset();
                    if (_hours.Count == 23)
                    {
                        _hours.Reset();
                    }
                    else
                    {
                        _minutes.Increment();

                    }
                }
            }
        }
    }
}

File: 10.1P/clock.js
class Counter {
	value = 0;
	constructor(n) {
		this.name = n;
	}

	get Name() {
		return this.name;
	}

	get Count() {
		return this.value;
	}
	Increment() {
		this.value++;
	}
	Reset() {
		this.value = 0;
	}
}

class Clock {
	seconds = new Counter('seconds');
	minutes = new Counter('minutes');
	hours = new Counter('hours');

	Reset() {
		this.seconds.Reset();
		this.minutes.Reset();
		this.hours.Reset();
	}

	get ReadTime() {
		return `${this.hours.Count}:${this.minutes.Count}:${this.seconds.Count}`;
	}

	Tick() {
		if (this.seconds.Count < 59) {
			this.seconds.Increment();
		} else {
			this.seconds.Reset();
			if (this.minutes.Count < 59) {
				this.minutes.Increment();
			} else {
				this.minutes.Reset();
				if (this.hours.Count == 23) {
					this.hours.Reset();
				} else {
					this.hours.Increment();
				}
			}
		}
	}
}

let clock = new Clock();

for (var i = 0; i < 60 * 60 * 24 + 61; i++) {
	clock.Tick();
	console.log(clock.ReadTime);
}
